GX-TimeFrequency/src/minimax_grids.F90

```diff
@@ -0,0 +1,465 @@
+  !! @param[inout] weights: corresponding tranformation weights
+  !! @param[in] e_min: Minimum transition energy.
+  !! @param[in] e_max: Maximum transition energy.
+  !! @param[inout] max_errors: Max error for the three kind of transforms
+  !! @param[in] transformation type : 1 the cosine transform cos(it) -> cos(iw)
+  !!                                : 2 the cosine transform cos(iw) -> cos(it)
+  !!                                : 3 the sine transform   sin(it) -> sin(iw)
+  !! @param[in] ierr: exit status
+  subroutine get_transformation_weights(num_points, tau_points, omega_points, weights, e_min, e_max, &
+       max_error, transformation_type, ierr)
+
+    integer, intent(in)                                :: num_points
+    real(kind=dp), allocatable, dimension(:), &
+         intent(in)                                    :: tau_points
+    real(kind=dp), allocatable, dimension(:), &
+         intent(in)                                    :: omega_points
+    real(kind=dp), allocatable, dimension(:, :), &
+         intent(inout)                                 :: weights
+    real(kind=dp), intent(in)                          :: e_min, e_max
+    real(kind=dp), intent(inout)                       :: max_error
+    integer, intent(in)                                :: transformation_type
+    integer, intent(out)                               :: ierr
+
+    ! Internal variables
+    integer                                            :: i_node, i_point, j_point, k_point, &
+         num_x_nodes
+    integer, parameter                                 :: nodes_factor = 200
+    real(kind=dp)                                      :: current_point, x_factor
+    real(kind=dp), allocatable, dimension(:)           :: weights_work, x_mu, psi
+    real(kind=dp), allocatable, dimension(:, :)        :: mat_A
+
+    integer                                            :: lwork
+    integer, allocatable, dimension(:)                 :: iwork      
+    real(kind=dp), allocatable, dimension(:)           :: vec_S, vec_UT_psi, work     
+    real(kind=dp), allocatable, dimension(:, :)        :: mat_U, mat_VT, mat_VT_s
+
+    ! Begin work
+    ierr = 0
+
+    allocate (weights_work(num_points))
+    weights_work = 0.0_dp
+
+    ! compute the number of x nodes per magnitude points per 10-interval
+    num_x_nodes = (int(log10(e_max/e_min)) + 1)*nodes_factor
+
+    ! make sure that the number of x nodes are at least as many integration points
+    num_x_nodes = max(num_x_nodes, num_points)
+
+    allocate (x_mu(num_x_nodes))
+    x_mu = 0.0_dp
+    allocate (psi(num_x_nodes))
+    psi = 0.0_dp
+
+    ! Allocations for the BLAS routines
+    ! double the value nessary for 'A' to achieve good performance
+    lwork = 8*num_points*num_points + 12*num_points + 2*num_x_nodes
+    allocate (iwork(8*num_points))
+    iwork = 0
+    allocate (work(lwork))
+    work = 0.0_dp      
+
+    allocate (mat_A(num_x_nodes, num_points))
+    mat_A = 0.0_dp      
```

Comment:
Do these need to be initialised?

If so, you can save some space with:

`allocate (mat_A(num_x_nodes, num_points), source=0._dp)`

Reply:
Good to know.